### ISSUE TITLE
where use the relu fuction?

### ISSUE THREAD

**Issue by LinB203** [THUDM/ImageReward#33]
The paper algorithm says use relu for reward model to predict score. But there is no relu in your code. Anything I miss?
If do not use non-linear activation fuction, the multiple nn.Linear are equal to a single nn.Linear. For example, equal to nn.Linear(1024,1) in your settings.
![image](https://github.com/THUDM/ImageReward/assets/62638829/dab2932e-f9c6-47ce-8b10-9e14ecb94d7e)

**Comment by xujz18** [comment]
In fact, what paper says is that: "For ReFL algorithm, we set ϕ = ReLU" (Sec 4.2 ReFL...), there's no say "relu for reward model to predict score". See https://github.com/THUDM/ImageReward/blob/main/ImageReward/ReFL.py#L755 to find ReLU for ReFL.

**Comment by LinB203** [comment]
> In fact, what paper says is that: "For ReFL algorithm, we set ϕ = ReLU" (Sec 4.2 ReFL...), there's no say "relu for reward model to predict score". See https://github.com/THUDM/ImageReward/blob/main/ImageReward/ReFL.py#L755 to find ReLU for ReFL.

Sorry, my fault. And I wonder that why use so many nn.Linear? If do not use non-linear activation fuction, the multiple nn.Linear are equal to a single nn.Linear. For example, equal to nn.Linear(1024,1) in your settings.

**Comment by xujz18** [comment]
You can refer https://www.simplilearn.com/tutorials/deep-learning-tutorial/multilayer-perceptron to get a sense of Multi-layer ANN. The multiple nn.Linear are not equal to a single nn.Linear. The parameter matrix is different in both shape and number.

**Comment by LinB203** [comment]
> You can refer https://www.simplilearn.com/tutorials/deep-learning-tutorial/multilayer-perceptron to get a sense of Multi-layer ANN. The multiple nn.Linear are not equal to a single nn.Linear. The parameter matrix is different in both shape and number.

thanks a lot!
